### PR TITLE
Update to angr==9.2.77

### PR DESCRIPTION
**Pull request by ANogin**
- [X] I have reviewed the [OFRAK contributor guide](https://ofrak.com/docs/contributor-guide/getting-started.html) and attest that this pull request is in accordance with it.

**One sentence summary of this PR (This should go in the CHANGELOG!)**
Update to recent angr==9.2.77, which also necessitates Python >= 3.8 and capstone==5.0.0.post1

**Link to Related Issue(s)**
N/A

**Please describe the changes in your request.**

**Anyone you think should look at this, specifically?**
@EdwardLarson 

Note:
- This requires python 3.8, so includes and depends on #416 
- Two tests in `ofrak_angr` fail with this one

### CODE REVIEW

File: disassemblers/ofrak_angr/ofrak_angr_test/test_unpackers.py [hunk 1]
```diff
@@ -54,14 +54,30 @@ async def expected_results(self, unpack_verify_test_case: ComplexBlockUnpackerTe
...
+                keep_same_is_exit_point=False,
             )
 
             return self._fixup_test_case_for_pie(
                 unpack_verify_test_case.expected_results,
                 pie_base_vaddr=0x400000,
             )
 
+        elif unpack_verify_test_case.binary_md5_digest == "c79d1bea0398d7a9d0faa1ba68786f5e":
+            # Latest version of angr misses this DataWord now = the ref to it does not appear in the list of xrefs
```
Comment (whyitfor) [inline disassemblers/ofrak_angr/ofrak_angr_test/test_unpackers.py:66]: Can we specify which version specifically in this comment?

File: docs/getting-started.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 ## Quick Start - Unpack a firmware file and display it in the GUI
 
 !!! warning
-    OFRAK is a Python library supporting Python3.7 and up. First and foremost, make sure your Python and pip installations are for Python3.7+! Python 3.8 is recommended, as this is the version we primarily test OFRAK with.
+    OFRAK is a Python library supporting Python3.8 and up. First and foremost, make sure your Python and pip installations are for Python3.8+! Python 3.8 is recommended, as this is the version we primarily test OFRAK with.
```
Comment (whyitfor) [inline docs/getting-started.md:6]: Hm. Technically OFRAK still supports Python3.7; it is just `ofrak-angr` that requires 3.8 and up.

Are you suggesting that we drop 3.7 support with this commit? @rbs-jacob, would appreciate your thoughts on this. If we decide to do so, we should update all of the CHANGELOGs (and probably the "python_requires") in this repo.

**Review by whyitfor — COMMENTED** [review]
See comments. @rbs-jacob, your feedback would be appreciated as well.

File: disassemblers/ofrak_angr/ofrak_angr_test/test_unpackers.py (hunk 1 above)
Comment (ANogin) [reply disassemblers/ofrak_angr/ofrak_angr_test/test_unpackers.py]: Done.

File: docs/getting-started.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
 ## Quick Start - Unpack a firmware file and display it in the GUI
 
 !!! warning
-    OFRAK is a Python library supporting Python3.7 and up. First and foremost, make sure your Python and pip installations are for Python3.7+! Python 3.8 is recommended, as this is the version we primarily test OFRAK with.
+    OFRAK is a Python library supporting Python3.7 and up. First and foremost, make sure your Python and pip installations are for Python3.7+! Python 3.8 is recommended, as this is the version we primarily test OFRAK with, and some components require Python >=3.8.
```
Comment (whyitfor) [inline docs/getting-started.md:6]:
```suggestion
    OFRAK is a Python library supporting Python3.7 and up. First and foremost, make sure your Python and pip installations are for Python3.7+! Python 3.8 is recommended, as this is the version we primarily test OFRAK with, and some packages (for example, ofrak-angr) require Python >=3.8.
```

File: docs/environment-setup.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Environment Setup & Installing OFRAK
 
 !!! warning
-    OFRAK is a Python library supporting Python3.7 and up. First and foremost, make sure your Python and pip installations are for Python3.7+! Python 3.8 is recommended, as this is the version we primarily test OFRAK with.
+    OFRAK is a Python library supporting Python3.7 and up. First and foremost, make sure your Python and pip installations are for Python3.7+! Python 3.8 is recommended, as this is the version we primarily test OFRAK with, and some components require Python >=3.8.
```
Comment (whyitfor) [inline docs/environment-setup.md:4]:
```suggestion
    OFRAK is a Python library supporting Python3.7 and up. First and foremost, make sure your Python and pip installations are for Python3.7+! Python 3.8 is recommended, as this is the version we primarily test OFRAK with, and some packages (for example, ofrak-angr) require Python >=3.8.
```

**Review by whyitfor — APPROVED** [review]
LGTM

**Pull request merged**